blazar/plugins/base.py

```diff
@@ -158,6 +161,29 @@ def is_project_allowed(self, project_id, resource):
             return project_id in authorized_projects
         return True
 
+    def add_extra_allocation_info(self, resource_allocations):
+        """Add extra information to allocations (to show in calendar)"""
+        extras = CONF.manager.extras
+        for allocs in resource_allocations.values():
+            for alloc in allocs:
+                alloc["extras"] = []
+        if "name" in extras:
+            ids = []
+            for allocations in resource_allocations.values():
+                for alloc in allocations:
+                    ids.append(alloc["lease_id"])
+            items = db_utils.get_user_ids_for_lease_ids(ids)
+            keystoneclient = keystone.BlazarKeystoneClient()
+            lease_to_name = dict()
+            for lease_id, user_id in items:
+                user = keystoneclient.users.get(user_id)
+                lease_to_name[lease_id] = user.name
+
+            for allocations in resource_allocations.values():
+                for alloc in allocations:
+                    alloc["extras"] = [
+                        (_("Reserved by"), lease_to_name[alloc["lease_id"]]),
```

Comment:
I think it's better to return machine_names on endpoints like this. Horizon can be in charge of translating them to some proper UI text; as a GUI, this is its responsibility. Blazar shouldn't be making too many decisions about how data is displayed.